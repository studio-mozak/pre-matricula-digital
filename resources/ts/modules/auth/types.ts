export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
