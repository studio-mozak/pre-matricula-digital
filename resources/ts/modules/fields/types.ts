import { Nullable } from '@/types';

export type SetValues = (fields: Record<string, unknown>) => void;

export interface FieldPageOption {
  id?: string;
  name: Nullable<string>;
  weight: number;
}

export interface FieldPage {
  id: Nullable<string>;
  name: Nullable<string>;
  description: Nullable<string>;
  group: Nullable<'RESPONSIBLE' | 'STUDENT'>;
  type: Nullable<string>;
  internal: boolean;
  options: FieldPageOption[];
  deleteOptions?: string[];
  required?: boolean;
}

export interface ErrorResponse {
  message: string;
}
