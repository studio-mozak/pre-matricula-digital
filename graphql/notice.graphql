extend type Query @guard(with: ["web", "sanctum", "prematricula"]) {

  """
  Lista todos os avisos.
  """
  notices: [Notice!]! @paginate

  notice(id: ID @eq): Notice @find

}

extend type Mutation @guard {

  saveNotice(
    input: NoticeInput! @spread
  ): Notice!

  deleteNotice(
    id: ID! @whereKey
  ): Notice! @delete
}

type Notice {
  id: ID!
  text: String!
}

input NoticeInput {
  id: ID
  text: String!
}
