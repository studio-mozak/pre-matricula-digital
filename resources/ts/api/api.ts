import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
