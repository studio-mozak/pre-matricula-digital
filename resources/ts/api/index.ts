export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
