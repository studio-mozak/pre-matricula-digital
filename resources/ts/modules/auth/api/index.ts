/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
