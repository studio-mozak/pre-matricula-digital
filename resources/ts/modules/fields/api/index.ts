/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
