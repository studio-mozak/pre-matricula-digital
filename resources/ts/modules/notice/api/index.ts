/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Notice from './services/graphql/notice';
