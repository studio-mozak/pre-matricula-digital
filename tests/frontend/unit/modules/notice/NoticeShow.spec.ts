import { VueWrapper, flushPromises, shallowMount } from '@vue/test-utils';
import { Notice } from '@/modules/notice/types';
import { Notice as NoticeApi } from '@/modules/notice/api';
import NoticeShow from '@/modules/notice/components/NoticeShow.vue';
import { expect, vi } from 'vitest';

vi.mock('@/modules/notice/api');

const notice: Notice = {
  id: 1,
  text: 'test notice',
};

const onMounted = async (response: Notice) => {
  vi.mocked(NoticeApi.list).mockReturnValueOnce(Promise.resolve(response));

  const wrapper = shallowMount(NoticeShow);

  await flushPromises();

  await wrapper.vm.$nextTick();

  return wrapper;
};

describe('NoticeShow', () => {
  let wrapper: VueWrapper<InstanceType<typeof NoticeShow>>;

  beforeEach(async () => {
    wrapper = await onMounted(notice);

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.resetAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('notice value is successfully loaded', () => {
    expect(wrapper.vm.notice).toEqual(notice);
  });

  test('the value of notice is printed in the DOM', () => {
    const noticeDOM = wrapper.find('[data-test="notice-section"]');
    expect(noticeDOM.html()).toContain(notice.text);
  });

  test('notice value is empty', async () => {
    wrapper = await onMounted(undefined as unknown as Notice);

    expect(wrapper.vm.notice).toEqual({
      id: null,
      text: null,
    });
  });
});
