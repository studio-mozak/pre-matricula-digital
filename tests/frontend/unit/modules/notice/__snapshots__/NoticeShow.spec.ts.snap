// Vitest Snapshot v1

exports[`NoticeShow > should match snapshot 1`] = `
<div>
  <div
    class="row mt-3"
  >
    <header-entity-data-stub />
  </div>
  <div
    class="row"
  >
    <div
      class="col-12 col-lg-10 offset-lg-1"
    >
      <h2
        class="title-find-school"
      >
        Avisos
      </h2>
      <div
        class="mt-4"
      >
        <p
          data-test="notice-section"
        >
          test notice
        </p>
      </div>
    </div>
  </div>
</div>
`;
